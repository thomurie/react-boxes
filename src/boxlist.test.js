import { fireEvent, render, screen } from "@testing-library/react";
import Box from "./box";
import Boxlist from "./boxlist";

// Smoke
test("renders learn react link", () => {
  render(<Boxlist />);
});

// Snapshot
test("Matches snapshot", () => {
  const { asFragment } = render(<Boxlist />);
  expect(asFragment()).toMatchSnapshot();
});

// Critical Business Logic
test("Form renders", () => {
  const { getByLabelText, queryByText } = render(<Boxlist />);
  expect(getByLabelText("Width:")).toBeInTheDocument();
  expect(getByLabelText("Height:")).toBeInTheDocument();
  expect(getByLabelText("Color:")).toBeInTheDocument();
  expect(queryByText("Create Box")).toBeInTheDocument();
});

test("Submitting form renders box", () => {
  const { getByLabelText, queryByText, queryByTestId } = render(<Boxlist />);
  const widthInput = getByLabelText("Width:");
  const heightInput = getByLabelText("Height:");
  const colorInput = getByLabelText("Color:");
  const sumbitBtn = queryByText("Create Box");

  fireEvent.change(widthInput, { target: { value: "200" } });
  fireEvent.change(heightInput, { target: { value: "200" } });
  fireEvent.change(colorInput, { target: { value: "red" } });
  fireEvent.click(sumbitBtn);

  expect(queryByTestId("red")).toBeInTheDocument();
});

test("Can remove box from list", () => {
  // render component
  const { getByLabelText, queryByText, queryByTestId } = render(<Boxlist />);
  // create a box with the form
  const widthInput = getByLabelText("Width:");
  const heightInput = getByLabelText("Height:");
  const colorInput = getByLabelText("Color:");
  const sumbitBtn = queryByText("Create Box");
  // enter data in the form
  fireEvent.change(widthInput, { target: { value: "200" } });
  fireEvent.change(heightInput, { target: { value: "200" } });
  fireEvent.change(colorInput, { target: { value: "red" } });
  fireEvent.click(sumbitBtn);

  expect(queryByTestId("red")).toBeInTheDocument();
  // remove the box from the form
  const removeBox = queryByText("X");
  fireEvent.click(removeBox);
  expect(queryByTestId("red")).not.toBeInTheDocument();
});
