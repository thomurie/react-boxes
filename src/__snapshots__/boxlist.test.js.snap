// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewBoxForm"
  >
    <form>
      <div>
        <label
          for="width"
        >
          Width: 
        </label>
        <input
          id="width"
          name="width"
          placeholder="100-500"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="height"
        >
          Height: 
        </label>
        <input
          id="height"
          name="height"
          placeholder="100-500"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="color"
        >
          Color: 
        </label>
        <input
          id="color"
          name="color"
          placeholder="blue"
          type="text"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Create Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
